ts/spi/provider_frame.ts

```diff
@@ -344,6 +354,22 @@ export class ProviderFrame {
     return this.handleUnimplementedRequest(id, msg.RPC_MESSAGE_TYPES.proxy);
   }
 
+  private async handleCancelLastOperation(id: string) {
+    if (!this.requestInProgress || this.cancellable === null) {
+      // no request in progress
+      console.warn('No pending request to cancel.');
+      this.clientChannel.send(msg.errorMessage(
+          id, OpenYoloError.requestFailed('No request in progress.')));
```

Comment:
I would perhaps just treat this as a no-op; I think your response here will result in the cancel promise being rejected, which could have consequences for the client code.